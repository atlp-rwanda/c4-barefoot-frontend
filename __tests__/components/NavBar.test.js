import React from 'react';
import Header from '../../src/components/NavBar';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';


describe('<Header />', () =>{
    let wrapper;
    it("should be rendered without errors", () =>{
        wrapper = mount(<Header/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})