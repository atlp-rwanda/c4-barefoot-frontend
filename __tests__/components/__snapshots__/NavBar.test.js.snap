// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should be rendered without errors 1`] = `
<Header>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <ForwardRef(AppBar)
      classes={
        Object {
          "colorDefault": "MuiAppBar-colorDefault",
          "colorInherit": "MuiAppBar-colorInherit",
          "colorPrimary": "MuiAppBar-colorPrimary",
          "colorSecondary": "MuiAppBar-colorSecondary",
          "colorTransparent": "MuiAppBar-colorTransparent",
          "positionAbsolute": "MuiAppBar-positionAbsolute",
          "positionFixed": "MuiAppBar-positionFixed",
          "positionRelative": "MuiAppBar-positionRelative",
          "positionStatic": "MuiAppBar-positionStatic",
          "positionSticky": "MuiAppBar-positionSticky",
          "root": "MuiAppBar-root",
        }
      }
      position="static"
    >
      <WithStyles(ForwardRef(Paper))
        className="MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
        component="header"
        elevation={4}
        square={true}
      >
        <ForwardRef(Paper)
          className="MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
          classes={
            Object {
              "elevation0": "MuiPaper-elevation0",
              "elevation1": "MuiPaper-elevation1",
              "elevation10": "MuiPaper-elevation10",
              "elevation11": "MuiPaper-elevation11",
              "elevation12": "MuiPaper-elevation12",
              "elevation13": "MuiPaper-elevation13",
              "elevation14": "MuiPaper-elevation14",
              "elevation15": "MuiPaper-elevation15",
              "elevation16": "MuiPaper-elevation16",
              "elevation17": "MuiPaper-elevation17",
              "elevation18": "MuiPaper-elevation18",
              "elevation19": "MuiPaper-elevation19",
              "elevation2": "MuiPaper-elevation2",
              "elevation20": "MuiPaper-elevation20",
              "elevation21": "MuiPaper-elevation21",
              "elevation22": "MuiPaper-elevation22",
              "elevation23": "MuiPaper-elevation23",
              "elevation24": "MuiPaper-elevation24",
              "elevation3": "MuiPaper-elevation3",
              "elevation4": "MuiPaper-elevation4",
              "elevation5": "MuiPaper-elevation5",
              "elevation6": "MuiPaper-elevation6",
              "elevation7": "MuiPaper-elevation7",
              "elevation8": "MuiPaper-elevation8",
              "elevation9": "MuiPaper-elevation9",
              "outlined": "MuiPaper-outlined",
              "root": "MuiPaper-root",
              "rounded": "MuiPaper-rounded",
            }
          }
          component="header"
          elevation={4}
          square={true}
        >
          <header
            className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
          >
            <div>
              <div
                className="navbar"
              >
                <div
                  className="logo"
                >
                  <h2>
                    Barefoot Nomad
                  </h2>
                </div>
                <div
                  className="navlinks"
                >
                  <ul>
                    <li>
                      <WithStyles(ForwardRef(Link))
                        to="/"
                      >
                        <ForwardRef(Link)
                          classes={
                            Object {
                              "button": "MuiLink-button",
                              "focusVisible": "Mui-focusVisible",
                              "root": "MuiLink-root",
                              "underlineAlways": "MuiLink-underlineAlways",
                              "underlineHover": "MuiLink-underlineHover",
                              "underlineNone": "MuiLink-underlineNone",
                            }
                          }
                          to="/"
                        >
                          <WithStyles(ForwardRef(Typography))
                            className="MuiLink-root MuiLink-underlineHover"
                            color="primary"
                            component="a"
                            onBlur={[Function]}
                            onFocus={[Function]}
                            to="/"
                            variant="inherit"
                          >
                            <ForwardRef(Typography)
                              className="MuiLink-root MuiLink-underlineHover"
                              classes={
                                Object {
                                  "alignCenter": "MuiTypography-alignCenter",
                                  "alignJustify": "MuiTypography-alignJustify",
                                  "alignLeft": "MuiTypography-alignLeft",
                                  "alignRight": "MuiTypography-alignRight",
                                  "body1": "MuiTypography-body1",
                                  "body2": "MuiTypography-body2",
                                  "button": "MuiTypography-button",
                                  "caption": "MuiTypography-caption",
                                  "colorError": "MuiTypography-colorError",
                                  "colorInherit": "MuiTypography-colorInherit",
                                  "colorPrimary": "MuiTypography-colorPrimary",
                                  "colorSecondary": "MuiTypography-colorSecondary",
                                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                  "displayBlock": "MuiTypography-displayBlock",
                                  "displayInline": "MuiTypography-displayInline",
                                  "gutterBottom": "MuiTypography-gutterBottom",
                                  "h1": "MuiTypography-h1",
                                  "h2": "MuiTypography-h2",
                                  "h3": "MuiTypography-h3",
                                  "h4": "MuiTypography-h4",
                                  "h5": "MuiTypography-h5",
                                  "h6": "MuiTypography-h6",
                                  "noWrap": "MuiTypography-noWrap",
                                  "overline": "MuiTypography-overline",
                                  "paragraph": "MuiTypography-paragraph",
                                  "root": "MuiTypography-root",
                                  "srOnly": "MuiTypography-srOnly",
                                  "subtitle1": "MuiTypography-subtitle1",
                                  "subtitle2": "MuiTypography-subtitle2",
                                }
                              }
                              color="primary"
                              component="a"
                              onBlur={[Function]}
                              onFocus={[Function]}
                              to="/"
                              variant="inherit"
                            >
                              <a
                                className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                                onBlur={[Function]}
                                onFocus={[Function]}
                                to="/"
                              >
                                Home
                              </a>
                            </ForwardRef(Typography)>
                          </WithStyles(ForwardRef(Typography))>
                        </ForwardRef(Link)>
                      </WithStyles(ForwardRef(Link))>
                    </li>
                    <li>
                       
                      <WithStyles(ForwardRef(Link))
                        to="/login"
                      >
                        <ForwardRef(Link)
                          classes={
                            Object {
                              "button": "MuiLink-button",
                              "focusVisible": "Mui-focusVisible",
                              "root": "MuiLink-root",
                              "underlineAlways": "MuiLink-underlineAlways",
                              "underlineHover": "MuiLink-underlineHover",
                              "underlineNone": "MuiLink-underlineNone",
                            }
                          }
                          to="/login"
                        >
                          <WithStyles(ForwardRef(Typography))
                            className="MuiLink-root MuiLink-underlineHover"
                            color="primary"
                            component="a"
                            onBlur={[Function]}
                            onFocus={[Function]}
                            to="/login"
                            variant="inherit"
                          >
                            <ForwardRef(Typography)
                              className="MuiLink-root MuiLink-underlineHover"
                              classes={
                                Object {
                                  "alignCenter": "MuiTypography-alignCenter",
                                  "alignJustify": "MuiTypography-alignJustify",
                                  "alignLeft": "MuiTypography-alignLeft",
                                  "alignRight": "MuiTypography-alignRight",
                                  "body1": "MuiTypography-body1",
                                  "body2": "MuiTypography-body2",
                                  "button": "MuiTypography-button",
                                  "caption": "MuiTypography-caption",
                                  "colorError": "MuiTypography-colorError",
                                  "colorInherit": "MuiTypography-colorInherit",
                                  "colorPrimary": "MuiTypography-colorPrimary",
                                  "colorSecondary": "MuiTypography-colorSecondary",
                                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                  "displayBlock": "MuiTypography-displayBlock",
                                  "displayInline": "MuiTypography-displayInline",
                                  "gutterBottom": "MuiTypography-gutterBottom",
                                  "h1": "MuiTypography-h1",
                                  "h2": "MuiTypography-h2",
                                  "h3": "MuiTypography-h3",
                                  "h4": "MuiTypography-h4",
                                  "h5": "MuiTypography-h5",
                                  "h6": "MuiTypography-h6",
                                  "noWrap": "MuiTypography-noWrap",
                                  "overline": "MuiTypography-overline",
                                  "paragraph": "MuiTypography-paragraph",
                                  "root": "MuiTypography-root",
                                  "srOnly": "MuiTypography-srOnly",
                                  "subtitle1": "MuiTypography-subtitle1",
                                  "subtitle2": "MuiTypography-subtitle2",
                                }
                              }
                              color="primary"
                              component="a"
                              onBlur={[Function]}
                              onFocus={[Function]}
                              to="/login"
                              variant="inherit"
                            >
                              <a
                                className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                                onBlur={[Function]}
                                onFocus={[Function]}
                                to="/login"
                              >
                                Login
                              </a>
                            </ForwardRef(Typography)>
                          </WithStyles(ForwardRef(Typography))>
                        </ForwardRef(Link)>
                      </WithStyles(ForwardRef(Link))>
                    </li>
                    <li>
                      <WithStyles(ForwardRef(Link))
                        to="/signup"
                      >
                        <ForwardRef(Link)
                          classes={
                            Object {
                              "button": "MuiLink-button",
                              "focusVisible": "Mui-focusVisible",
                              "root": "MuiLink-root",
                              "underlineAlways": "MuiLink-underlineAlways",
                              "underlineHover": "MuiLink-underlineHover",
                              "underlineNone": "MuiLink-underlineNone",
                            }
                          }
                          to="/signup"
                        >
                          <WithStyles(ForwardRef(Typography))
                            className="MuiLink-root MuiLink-underlineHover"
                            color="primary"
                            component="a"
                            onBlur={[Function]}
                            onFocus={[Function]}
                            to="/signup"
                            variant="inherit"
                          >
                            <ForwardRef(Typography)
                              className="MuiLink-root MuiLink-underlineHover"
                              classes={
                                Object {
                                  "alignCenter": "MuiTypography-alignCenter",
                                  "alignJustify": "MuiTypography-alignJustify",
                                  "alignLeft": "MuiTypography-alignLeft",
                                  "alignRight": "MuiTypography-alignRight",
                                  "body1": "MuiTypography-body1",
                                  "body2": "MuiTypography-body2",
                                  "button": "MuiTypography-button",
                                  "caption": "MuiTypography-caption",
                                  "colorError": "MuiTypography-colorError",
                                  "colorInherit": "MuiTypography-colorInherit",
                                  "colorPrimary": "MuiTypography-colorPrimary",
                                  "colorSecondary": "MuiTypography-colorSecondary",
                                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                  "displayBlock": "MuiTypography-displayBlock",
                                  "displayInline": "MuiTypography-displayInline",
                                  "gutterBottom": "MuiTypography-gutterBottom",
                                  "h1": "MuiTypography-h1",
                                  "h2": "MuiTypography-h2",
                                  "h3": "MuiTypography-h3",
                                  "h4": "MuiTypography-h4",
                                  "h5": "MuiTypography-h5",
                                  "h6": "MuiTypography-h6",
                                  "noWrap": "MuiTypography-noWrap",
                                  "overline": "MuiTypography-overline",
                                  "paragraph": "MuiTypography-paragraph",
                                  "root": "MuiTypography-root",
                                  "srOnly": "MuiTypography-srOnly",
                                  "subtitle1": "MuiTypography-subtitle1",
                                  "subtitle2": "MuiTypography-subtitle2",
                                }
                              }
                              color="primary"
                              component="a"
                              onBlur={[Function]}
                              onFocus={[Function]}
                              to="/signup"
                              variant="inherit"
                            >
                              <a
                                className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                                onBlur={[Function]}
                                onFocus={[Function]}
                                to="/signup"
                              >
                                Signup
                              </a>
                            </ForwardRef(Typography)>
                          </WithStyles(ForwardRef(Typography))>
                        </ForwardRef(Link)>
                      </WithStyles(ForwardRef(Link))>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </header>
        </ForwardRef(Paper)>
      </WithStyles(ForwardRef(Paper))>
    </ForwardRef(AppBar)>
  </WithStyles(ForwardRef(AppBar))>
</Header>
`;
